public/app/plugins/datasource/cloudwatch/components/LogGroupSelector.tsx

```diff
@@ -113,7 +113,12 @@ export const LogGroupSelector: React.FC<LogGroupSelectorProps> = ({
         .then((logGroups) => {
           const newSelectedLogGroups = intersection(
             selectedLogGroups,
-            logGroups.map(({ value }) => value).filter((value): value is string => !!value)
+            logGroups.reduce((acc, { value }) => {
+              if (value) {
+                return [...acc, value];
+              }
+              return acc;
+            }, [] as string[])
```

Comment:
a `reduce` is even more difficult to read, and you are also using a casting, which should be avoided. Since in reality, value will be always populated, to satisfy typescript you can do:

```ts
          const newSelectedLogGroups = intersection(
            selectedLogGroups,
            logGroups.map(l => l.value || ''),
          );
```